### PR TITLE
Increase 512kbytes object put limit to 2mbytes

### LINKED ISSUE #2938: Increase 512 kbyte limit for `ipfs object put`
For a mirroring project, I'm trying to add very large folders to ipfs (> 10k links).

I've used `ipfs object patch add-link` for this, but switched to `ipfs object put` for performance reasons.

Current go-ipfs has a hardcoded limit for `ipfs object put`, which is [512 kbytes][1].

The folder I'm trying to add to ipfs is too large for this limit.

[1]: https://github.com/ipfs/go-ipfs/blob/ddc8d0c60c7e4e46f63afc3d754f4c973ae084a8/core/commands/object/object.go#L24

### PR DESCRIPTION
**Pull request by kpcyrd**
This resolves #2938

License: MIT
Signed-off-by: kpcyrd <git@rxv.cc>

### CODE REVIEW

File: core/commands/object/object.go [hunk 1]
```diff
@@ -21,10 +21,10 @@ import (
 	ft "github.com/ipfs/go-ipfs/unixfs"
 )
 
-// ErrObjectTooLarge is returned when too much data was read from stdin. current limit 512k
-var ErrObjectTooLarge = errors.New("input object was too large. limit is 512kbytes")
+// ErrObjectTooLarge is returned when too much data was read from stdin. current limit 2m
+var ErrObjectTooLarge = errors.New("input object was too large. limit is 2mbytes")
 
-const inputLimit = 512 * 1024
```
Comment (whyrusleeping) [inline core/commands/object/object.go]: nitpick: I want to move towards using `2 << 20` for things like this. The go team does this on all go projects and it makes it really easy to read, ex:

```
1 << 10 == 1K
4 << 10 == 4K
2 << 20 == 2MB
4 << 30 == 4GB
```

Comment (lgierth) [inline core/commands/object/object.go]: Agreed, although important to note it's Ki, Mi, and Gi :)

**Comment by kpcyrd** [comment]
Pushed a new patch

**Comment by whyrusleeping** [comment]
LGTM, thanks @kpcyrd !

**Pull request merged**